File: q3/JAVA/q3-1.java
public class Solution {
    public ArrayList<Integer> printListFromTailToHead(ListNode listNode) {
        ArrayList<Integer> result = new ArrayList<Integer>();
        ListNode pNode = listNode;
        
        Stack<Integer> myStack = new Stack<Integer>();
        while (pNode != null) {
            myStack.push(pNode.val);
            pNode = pNode.next;
        }
        
        while (!myStack.isEmpty()) {
            result.add(myStack.pop());
        }
        return result;
    }
}

File: q3/C++/q3-1.cpp
class Solution {
public:
    vector<int> printListFromTailToHead(ListNode* head) {
        stack<int> myStack;
        vector<int> result;
        
        ListNode* pNode = head;
        while (pNode != nullptr) {
            myStack.push(pNode->val);
            pNode = pNode->next;
        }
        
        while (!myStack.empty()) {
            result.push_back(myStack.top());
            myStack.pop();
        }
        
        return result;
    }
};
